tests/__main__.py

```diff
@@ -16,5 +16,5 @@
 
 params = " ".join(sys.argv[1:])
 
-cmd = "pytest -v -n=4 --cov=dvc --durations=0 {params}".format(params=params)
+cmd = "pytest -v -n=auto --dist loadscope --cov=dvc --durations=0 {params}".format(params=params)
```

Comment:
`loadscope` or `loadfile` reduces test time for me from 3.5min to 2.5min. GHA tests also save a few minutes (~1minute for mac and linux)